core/src/main/java/io/substrait/expression/ExpressionCreator.java

```diff
@@ -286,6 +287,33 @@ public static Expression.StructLiteral struct(
     return Expression.StructLiteral.builder().nullable(nullable).addAllFields(values).build();
   }
 
+  public static Expression.StructNested nestedStruct(
+      boolean nullable, Iterable<Expression> fields) {
+    return Expression.StructNested.builder().nullable(nullable).addAllFields(fields).build();
+  }
+
+  public static Expression.StructNested nestedStruct(boolean nullable, Expression... fields) {
+    return Expression.StructNested.builder()
+        .nullable(nullable)
+        .addAllFields(Arrays.asList(fields))
```

Comment:
minor: You can just use the following
```suggestion
        .addFields(fields)
```

to avoid having to construct an ArrayList.